### PR TITLE
improve react usage

### PR DESCRIPTION
**Pull request by stas-vilchik**

### CODE REVIEW

File: server/sonar-web/src/main/js/components/shared/drilldown-link.js [hunk 1]
```diff
@@ -47,12 +47,12 @@ const ISSUE_MEASURES = [
   'new_vulnerabilities'
 ];
 
-export const DrilldownLink = React.createClass({
-  isIssueMeasure() {
+export class DrilldownLink extends React.PureComponent {
+  isIssueMeasure = () => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/shared/drilldown-link.js]: MINOR Remove curly braces and "return" from this arrow function body. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/apps/projects-admin/search.js [hunk 2]
```diff
@@ -105,11 +105,11 @@ export default React.createClass({
         />
       </td>
     );
-  },
+  };
 
-  renderSpinner() {
+  renderSpinner = () => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/projects-admin/search.js]: MINOR Remove curly braces and "return" from this arrow function body. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/apps/system/item-log-level.js [hunk 3]
```diff
@@ -23,17 +23,15 @@ import { translate } from '../../helpers/l10n';
 
 const LOG_LEVELS = ['INFO', 'DEBUG', 'TRACE'];
 
-export default React.createClass({
-  getInitialState() {
-    return { level: this.props.value };
-  },
+export default class ItemLovLevel extends React.PureComponent {
+  state = { level: this.props.value };
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/system/item-log-level.js]: MINOR Remove the use of "this". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS2990)

File: server/sonar-web/src/main/js/apps/system/main.js [hunk 4]
```diff
@@ -36,36 +36,36 @@ const SECTIONS_ORDER = [
...
+  parseSections = data => {
     const sections = Object.keys(data).map(section => {
       return { name: section, items: this.parseItems(data[section]) };
     });
     return this.orderSections(sections);
-  },
+  };
 
-  orderSections(sections) {
+  orderSections = sections => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/system/main.js]: MINOR Remove curly braces and "return" from this arrow function body. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/apps/system/main.js [hunk 5]
```diff
@@ -36,36 +36,36 @@ const SECTIONS_ORDER = [
...
+  parseItems = data => {
     const items = Object.keys(data).map(item => {
       return { name: item, value: data[item] };
     });
     return this.orderItems(items);
-  },
+  };
 
-  orderItems(items) {
+  orderItems = items => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/system/main.js]: MINOR Remove curly braces and "return" from this arrow function body. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

**Comment by SonarTech** [comment]
SonarQube analysis reported 19 issues
* ![MINOR][MINOR] 19 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [main.js](https://github.com/SonarSource/sonarqube/blob/f7e4ff7038ff296247d31fd8b3864ea14854bc89/server/sonar-web/src/main/js/apps/projects-admin/main.js): Rename this file to "Main". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [search.js](https://github.com/SonarSource/sonarqube/blob/f7e4ff7038ff296247d31fd8b3864ea14854bc89/server/sonar-web/src/main/js/apps/projects-admin/search.js): Rename this file to "Search". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-boolean.js](https://github.com/SonarSource/sonarqube/blob/f7e4ff7038ff296247d31fd8b3864ea14854bc89/server/sonar-web/src/main/js/apps/system/item-boolean.js): Rename this file to "ItemBoolean". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-log-level.js](https://github.com/SonarSource/sonarqube/blob/f7e4ff7038ff296247d31fd8b3864ea14854bc89/server/sonar-web/src/main/js/apps/system/item-log-level.js): Rename this file to "ItemLovLevel". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-object.js](https://github.com/SonarSource/sonarqube/blob/f7e4ff7038ff296247d31fd8b3864ea14854bc89/server/sonar-web/src/main/js/apps/system/item-object.js): Rename this file to "ItemObject". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-value.js](https://github.com/SonarSource/sonarqube/blob/f7e4ff7038ff296247d31fd8b3864ea14854bc89/server/sonar-web/src/main/js/apps/system/item-value.js): Rename this file to "ItemValue". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [main.js](https://github.com/SonarSource/sonarqube/blob/f7e4ff7038ff296247d31fd8b3864ea14854bc89/server/sonar-web/src/main/js/apps/system/main.js): Rename this file to "Main". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [section.js](https://github.com/SonarSource/sonarqube/blob/f7e4ff7038ff296247d31fd8b3864ea14854bc89/server/sonar-web/src/main/js/apps/system/section.js): Rename this file to "Section". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [controls.js](https://github.com/SonarSource/sonarqube/blob/f7e4ff7038ff296247d31fd8b3864ea14854bc89/server/sonar-web/src/main/js/components/select-list/controls.js): Rename this file to "Controls". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [footer.js](https://github.com/SonarSource/sonarqube/blob/f7e4ff7038ff296247d31fd8b3864ea14854bc89/server/sonar-web/src/main/js/components/select-list/footer.js): Rename this file to "Footer". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 17 issues
* ![MINOR][MINOR] 17 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [main.js](https://github.com/SonarSource/sonarqube/blob/2152c017630eaeccd16df99cea2cf0f61f5241fc/server/sonar-web/src/main/js/apps/projects-admin/main.js): Rename this file to "Main". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [search.js](https://github.com/SonarSource/sonarqube/blob/2152c017630eaeccd16df99cea2cf0f61f5241fc/server/sonar-web/src/main/js/apps/projects-admin/search.js): Rename this file to "Search". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-boolean.js](https://github.com/SonarSource/sonarqube/blob/2152c017630eaeccd16df99cea2cf0f61f5241fc/server/sonar-web/src/main/js/apps/system/item-boolean.js): Rename this file to "ItemBoolean". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-log-level.js](https://github.com/SonarSource/sonarqube/blob/2152c017630eaeccd16df99cea2cf0f61f5241fc/server/sonar-web/src/main/js/apps/system/item-log-level.js): Rename this file to "ItemLovLevel". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-object.js](https://github.com/SonarSource/sonarqube/blob/2152c017630eaeccd16df99cea2cf0f61f5241fc/server/sonar-web/src/main/js/apps/system/item-object.js): Rename this file to "ItemObject". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-value.js](https://github.com/SonarSource/sonarqube/blob/2152c017630eaeccd16df99cea2cf0f61f5241fc/server/sonar-web/src/main/js/apps/system/item-value.js): Rename this file to "ItemValue". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [main.js](https://github.com/SonarSource/sonarqube/blob/2152c017630eaeccd16df99cea2cf0f61f5241fc/server/sonar-web/src/main/js/apps/system/main.js): Rename this file to "Main". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [section.js](https://github.com/SonarSource/sonarqube/blob/2152c017630eaeccd16df99cea2cf0f61f5241fc/server/sonar-web/src/main/js/apps/system/section.js): Rename this file to "Section". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [controls.js](https://github.com/SonarSource/sonarqube/blob/2152c017630eaeccd16df99cea2cf0f61f5241fc/server/sonar-web/src/main/js/components/select-list/controls.js): Rename this file to "Controls". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [footer.js](https://github.com/SonarSource/sonarqube/blob/2152c017630eaeccd16df99cea2cf0f61f5241fc/server/sonar-web/src/main/js/components/select-list/footer.js): Rename this file to "Footer". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 13 issues
* ![MINOR][MINOR] 13 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [main.js](https://github.com/SonarSource/sonarqube/blob/e56cc9378156d6b53148a9eb7b5c5f6a5f534915/server/sonar-web/src/main/js/apps/projects-admin/main.js): Rename this file to "Main". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [search.js](https://github.com/SonarSource/sonarqube/blob/e56cc9378156d6b53148a9eb7b5c5f6a5f534915/server/sonar-web/src/main/js/apps/projects-admin/search.js): Rename this file to "Search". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-boolean.js](https://github.com/SonarSource/sonarqube/blob/e56cc9378156d6b53148a9eb7b5c5f6a5f534915/server/sonar-web/src/main/js/apps/system/item-boolean.js): Rename this file to "ItemBoolean". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-log-level.js](https://github.com/SonarSource/sonarqube/blob/e56cc9378156d6b53148a9eb7b5c5f6a5f534915/server/sonar-web/src/main/js/apps/system/item-log-level.js): Rename this file to "ItemLovLevel". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-object.js](https://github.com/SonarSource/sonarqube/blob/e56cc9378156d6b53148a9eb7b5c5f6a5f534915/server/sonar-web/src/main/js/apps/system/item-object.js): Rename this file to "ItemObject". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-value.js](https://github.com/SonarSource/sonarqube/blob/e56cc9378156d6b53148a9eb7b5c5f6a5f534915/server/sonar-web/src/main/js/apps/system/item-value.js): Rename this file to "ItemValue". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [main.js](https://github.com/SonarSource/sonarqube/blob/e56cc9378156d6b53148a9eb7b5c5f6a5f534915/server/sonar-web/src/main/js/apps/system/main.js): Rename this file to "Main". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [section.js](https://github.com/SonarSource/sonarqube/blob/e56cc9378156d6b53148a9eb7b5c5f6a5f534915/server/sonar-web/src/main/js/apps/system/section.js): Rename this file to "Section". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [controls.js](https://github.com/SonarSource/sonarqube/blob/e56cc9378156d6b53148a9eb7b5c5f6a5f534915/server/sonar-web/src/main/js/components/select-list/controls.js): Rename this file to "Controls". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [footer.js](https://github.com/SonarSource/sonarqube/blob/e56cc9378156d6b53148a9eb7b5c5f6a5f534915/server/sonar-web/src/main/js/components/select-list/footer.js): Rename this file to "Footer". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: server/sonar-web/src/main/js/apps/system/item-log-level.js [hunk 6]
```diff
@@ -23,17 +23,15 @@ import { translate } from '../../helpers/l10n';
 
 const LOG_LEVELS = ['INFO', 'DEBUG', 'TRACE'];
 
-export default React.createClass({
-  getInitialState() {
-    return { level: this.props.value };
-  },
+export default class ItemLovLevel extends React.PureComponent {
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/system/item-log-level.js]: Typo : should be `ItemLogLevel`

File: server/sonar-web/src/main/js/components/common/BubblePopupHelper.js [hunk 7]
```diff
@@ -22,13 +22,13 @@ import classNames from 'classnames';
 
 type Props = {
   className?: string,
-  children: React.Component<*>,
+  children?: Object,
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/components/common/BubblePopupHelper.js]: Why switch from `Component` to `Object` ?

**Comment by SonarTech** [comment]
SonarQube analysis reported 12 issues
* ![MINOR][MINOR] 12 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [main.js](https://github.com/SonarSource/sonarqube/blob/cd74772c61d18a0b048a49f22592ae32967031a6/server/sonar-web/src/main/js/apps/projects-admin/main.js): Rename this file to "Main". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [search.js](https://github.com/SonarSource/sonarqube/blob/cd74772c61d18a0b048a49f22592ae32967031a6/server/sonar-web/src/main/js/apps/projects-admin/search.js): Rename this file to "Search". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-boolean.js](https://github.com/SonarSource/sonarqube/blob/cd74772c61d18a0b048a49f22592ae32967031a6/server/sonar-web/src/main/js/apps/system/item-boolean.js): Rename this file to "ItemBoolean". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-log-level.js](https://github.com/SonarSource/sonarqube/blob/cd74772c61d18a0b048a49f22592ae32967031a6/server/sonar-web/src/main/js/apps/system/item-log-level.js): Rename this file to "ItemLogLevel". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-object.js](https://github.com/SonarSource/sonarqube/blob/cd74772c61d18a0b048a49f22592ae32967031a6/server/sonar-web/src/main/js/apps/system/item-object.js): Rename this file to "ItemObject". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [item-value.js](https://github.com/SonarSource/sonarqube/blob/cd74772c61d18a0b048a49f22592ae32967031a6/server/sonar-web/src/main/js/apps/system/item-value.js): Rename this file to "ItemValue". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [main.js](https://github.com/SonarSource/sonarqube/blob/cd74772c61d18a0b048a49f22592ae32967031a6/server/sonar-web/src/main/js/apps/system/main.js): Rename this file to "Main". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [section.js](https://github.com/SonarSource/sonarqube/blob/cd74772c61d18a0b048a49f22592ae32967031a6/server/sonar-web/src/main/js/apps/system/section.js): Rename this file to "Section". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [controls.js](https://github.com/SonarSource/sonarqube/blob/cd74772c61d18a0b048a49f22592ae32967031a6/server/sonar-web/src/main/js/components/select-list/controls.js): Rename this file to "Controls". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)
1. ![MINOR][MINOR] [footer.js](https://github.com/SonarSource/sonarqube/blob/cd74772c61d18a0b048a49f22592ae32967031a6/server/sonar-web/src/main/js/components/select-list/footer.js): Rename this file to "Footer". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3317)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by gregaubert** [comment]
LGTM!

**Pull request merged**